assets/app.js

```diff
@@ -69,13 +69,36 @@ $(function () {
         });
     });
 
+    let selectLangField = $('#lang');
+    let previousDataPlaceholder = selectLangField.attr('data-placeholder');
+
     // Show engine-specific options.
     $('[name=engine]').on('change',  e => {
         updateSelect2Options(e.target.value);
         $('.engine-options').addClass('hidden');
         $(`#${e.target.value}-options`).removeClass('hidden');
+        let engine = e.target.value;
+        if(engine === 'tesseract' || engine === 'google') {
+            selectLangField.prop('required', false);
+            selectLangField.attr('data-placeholder', previousDataPlaceholder);
+            $('#transkribus-lang-label').addClass('hidden');
+            $('#optional-lang-label').removeClass('hidden');
+        } else {
+            selectLangField.prop('required', true);
+            selectLangField.attr('data-placeholder', '');
+            $('#optional-lang-label').addClass('hidden');
+            $('#transkribus-lang-label').removeClass('hidden');
+        }
     });
 
+    // modify selected engine after loading the page with preselected engine 
+    let engineRadioFields = $('[name=engine]:checked');
+    if(engineRadioFields.val() === 'transkribus') {
+        selectLangField.attr('data-placeholder', '')
+    } else {
+        selectLangField.attr('data-placeholder', previousDataPlaceholder);
+    }
```

Comment:
Rather than repeating these here, it might instead be possible to call `.trigger('change')` after the above `.on('change')` and so re-use the same code. But it's fine like this too, I'm sure.